src/core/constraint.rs

```diff
@@ -102,7 +103,13 @@ impl OrderConstraint<Expr> {
         }
     }
 
-    pub fn eval(self, binding: &Binding<Expr>) -> FilamentResult<bool> {
+    pub fn exprs(&self) -> Vec<&Expr> {
+        vec![&self.left, &self.right]
+    }
+}
+
+impl EvalBool for OrderConstraint<Expr> {
```

Comment:
The general recommendation from Rust is that if you're converting from type `A` to `B`, you can implement the `impl From<A> for B` which is a more general, known interface. Maybe it's worth implementing this as a 
```
impl From<OrderConstraint<Expr>> for bool { ... }
```

Reply:
Ah wait, this requires a binding to be resolved. NVM, it cannot be a `From` implementation